source/persistency/PetaloPersistencyManager.cc

```diff
@@ -306,22 +305,18 @@ void PetaloPersistencyManager::StoreSensorHits(G4VHitsCollection* hc)
     for (it = wvfm.begin(); it != wvfm.end(); ++it) {
       amplitude = amplitude + (*it).second;
     }
-    if (hit->GetPmtID() >= 0) {
-      G4int sens_id;
-      sens_id = hit->GetPmtID();
+    G4int sens_id;
+    sens_id = hit->GetSnsID();
```

Comment:
And here we don't need the loop because there are no charges in TOF so all sensor ids will be positive and, thus, from sns_response, right?

Reply:
Which loop? I can only see two lines in the diff.

Reply:
Not the loop but the if condition right before: `if (hit->GetPmtID() >= 0) {`.

Reply:
Yeah, now we can forget about negative sensor IDs, we have just positive ones.